src/Datadog.Trace.ClrProfiler.Managed/CallTarget/Handlers/IntegrationMapper.cs

```diff
@@ -0,0 +1,714 @@
+                    {
+                        ilWriter.Emit(OpCodes.Box, targetType);
+                    }
+
+                    ilWriter.Emit(OpCodes.Newobj, ctor);
+                }
+            }
+
+            // Load arguments
+            for (var i = mustLoadInstance ? 1 : 0; i < onMethodBeginParameters.Length; i++)
+            {
+                Type targetParameterType = onMethodBeginParameters[i].ParameterType;
+                Type targetParameterTypeConstraint = null;
+
+                if (targetParameterType.IsGenericParameter)
+                {
+                    targetParameterType = genericArgumentsTypes[targetParameterType.GenericParameterPosition];
+
+                    targetParameterTypeConstraint = targetParameterType.GetGenericParameterConstraints().FirstOrDefault(pType => pType != typeof(IDuckType));
+                    if (targetParameterTypeConstraint is null)
+                    {
+                        callGenericTypes.Add(typeof(object));
+                    }
+                    else
+                    {
+                        targetParameterType = targetParameterTypeConstraint;
+                        callGenericTypes.Add(targetParameterTypeConstraint);
+                    }
+                }
+
+                ilWriter.Emit(OpCodes.Ldarg_1);
+                WriteIntValue(ilWriter, i - (mustLoadInstance ? 1 : 0));
+                ilWriter.Emit(OpCodes.Ldelem_Ref);
+
+                if (targetParameterTypeConstraint != null)
+                {
+                    ilWriter.EmitCall(OpCodes.Call, ConvertTypeMethodInfo.MakeGenericMethod(targetParameterTypeConstraint), null);
+                }
+
+                if (targetParameterType.IsValueType)
+                {
+                    ilWriter.Emit(OpCodes.Unbox_Any, targetParameterType);
+                }
+            }
+
+            // Call method
+            // Log.Information("Generic Types: " + string.Join(", ", callGenericTypes.Select(t => t.FullName)));
+            // Log.Information("Method: " + onMethodBeginMethodInfo);
+            onMethodBeginMethodInfo = onMethodBeginMethodInfo.MakeGenericMethod(callGenericTypes.ToArray());
+            // Log.Information("Method: " + onMethodBeginMethodInfo);
+            ilWriter.EmitCall(OpCodes.Call, onMethodBeginMethodInfo, null);
+            ilWriter.Emit(OpCodes.Ret);
+
+            Log.Debug($"Created SlowBeginMethod Dynamic Method for '{integrationType.FullName}' integration. [Target={targetType.FullName}]");
+            return callMethod;
+        }
+
+        internal static DynamicMethod CreateEndMethodDelegate(Type integrationType, Type targetType)
+        {
+            Log.Debug($"Creating EndMethod Dynamic Method for '{integrationType.FullName}' integration. [Target={targetType.FullName}]");
+            MethodInfo onMethodEndMethodInfo = integrationType.GetMethod(EndAsyncMethodName, BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Static);
```

Comment:
Shouldn't the method name be `EndMethodName` not `EndAsyncMethodName`?

```suggestion
            MethodInfo onMethodEndMethodInfo = integrationType.GetMethod(EndAsyncMethodName, BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Static);
```

Reply:
yes in this method all should be EndMethodName, thanks to point it out, this is an error from the VS Rename :o

Reply:
fixed in https://github.com/DataDog/dd-trace-dotnet/pull/1007/commits/04ae61a340872f36f5e93929e69b26305bc991dd